sklearn/discriminant_analysis.py

```diff
@@ -43,29 +42,52 @@ def _cov(X, shrinkage=None):
           - 'auto': automatic shrinkage using the Ledoit-Wolf lemma.
           - float between 0 and 1: fixed shrinkage parameter.
 
+        Shrinkage parameter is ignored if  `covariance_estimator`
+        is not None.
+
+    covariance_estimator : estimator, default=None
+        If not None, `covariance_estimator` is used to estimate
+        the covariance matrices instead of relying the empirical
+        covariance estimator (with potential shrinkage).
+        The object should have a fit method and a ``covariance_`` attribute
+        like the estimators in sklearn.covariance.
+        if None the shrinkage parameter drives the estimate.
+
+        .. versionadded:: 0.23
+
     Returns
     -------
     s : ndarray of shape (n_features, n_features)
         Estimated covariance matrix.
     """
-    shrinkage = "empirical" if shrinkage is None else shrinkage
-    if isinstance(shrinkage, str):
-        if shrinkage == 'auto':
-            sc = StandardScaler()  # standardize features
-            X = sc.fit_transform(X)
-            s = ledoit_wolf(X)[0]
-            # rescale
-            s = sc.scale_[:, np.newaxis] * s * sc.scale_[np.newaxis, :]
-        elif shrinkage == 'empirical':
-            s = empirical_covariance(X)
+    if covariance_estimator is None:
```

Comment:
You confirm you didn't change anything to this bloc and only indented it to the right?

Reply:
Yes I do !